lib/InfluxDB/HTTP.pm

```diff
@@ -93,13 +93,30 @@ method query (Str|ArrayRef[Str] $query!, Str :$database, Int :$chunk_size, Str :
     }
 }
 
-method write (Str|ArrayRef[Str] $measurement!, Str :$database) {
+method write (Str|ArrayRef[Str] $measurement!, Str :$database, Str :$precision) {
```

Comment:
@niclan The following constraint should work (although rather long):
`Str :$precision where { ($_ == undef) || ($_ =~ /^(h|m|s|ms|u|ns)$/) }`